internal/cluster/clusteradapter/node_pool_validator.go

```diff
@@ -0,0 +1,119 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusteradapter
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+	"github.com/jinzhu/gorm"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/banzaicloud/pipeline/internal/cluster"
+	"github.com/banzaicloud/pipeline/internal/cluster/distribution"
+	"github.com/banzaicloud/pipeline/pkg/providers"
+	"github.com/banzaicloud/pipeline/src/model"
+)
+
+// DistributionNodePoolValidator validates a node pool request according to its own distribution.
+type DistributionNodePoolValidator struct {
+	db *gorm.DB
+}
+
+// NewDistributionNodePoolValidator returns a new DistributionNodePoolValidator.
+func NewDistributionNodePoolValidator(db *gorm.DB) DistributionNodePoolValidator {
+	return DistributionNodePoolValidator{
+		db: db,
+	}
+}
+
+func (v DistributionNodePoolValidator) Validate(_ context.Context, c cluster.Cluster, rawNodePool cluster.NewRawNodePool) error {
+	switch {
+	case c.Cloud == providers.Amazon && c.Distribution == "eks":
```

Comment:
```suggestion
	case c.Cloud == providers.Amazon && c.Distribution == cluster.EKS:
```

there are constants defined for distributions in `pkg/cluster/base.go`

Reply:
I don't want to import that package here for now.